### PR TITLE
Viewer update 2.10.3

### PR DESCRIPTION
**Pull request by Exilliar**
Updated the viewer to 2.10.3 . Fairly large update as there were lots of breaking changes to the viewer. I've tried as much as possible to only change the Renderer.vue component and make that component still take in the same inputs and emit the same outputs as it did before. As far as I can see everything is now working the same (or better) as it was before.

I did basically fully turn off the manual volume calculations for now. The new viewer changes up the way that you get the THREE.js objects (I'm not 100% sure if you can get them without doing a bit of a weird work-around), which seemed like it would be a fair bit more work to implement. I figured that as not many users are likely to be using that (and if they are we can just tell them to put some volume properties on their model), and I'm also not sure if it actually works on the live version currently anyway, it'd be fine to remove it for now.

I also removed the "quick report" mode as I thought that it looked a bit rubbish and also didn't really work well. The "edit" button now just takes the user straight to the full edit report page.

Also added in some lighting controls for the viewer (stolen from the Speckle server frontend) cuz it was fairly quick to do.

Also added in another loading spinner to the assessment page as there is some loading that needs to be done after the speckle model has been loaded that allows the user to enter in their materials. There is some text on the spinner to indicate what is happening.

Also fixed a bug on the "edit report" page where the volume didn't really seem to be working, the colours weren't appearing properly. Now it is (might've also just been caused by the viewer update).

There's also now a proper loading bar on the viewer.

The new viewer seems to work way better for larger models. I tested it with a couple and it loads fairly quickly and you can do reports pretty quickly as well.

I think that just about every page needs to be tested for this pr.

Leaving this as a draft pr for now as I want to do one final check on Monday.

EDIT:
While this pr was up a new version of the viewer was released. Updated to that, v2.11.4

### CODE REVIEW

File: src/components/shared/Renderer.vue [hunk 1]
```diff
@@ -50,90 +133,128 @@ export default class extends Vue {
...
   showAlert = false;
-  viewer!: any;
-  selectedObjects: any[] = [];
+  viewer!: Viewer;
 
   loading = 0;
   failed = false;
-  mounted() {
+  async mounted() {
     this.renderStream(this.objecturls);
```
Comment (ro-sa) [inline src/components/shared/Renderer.vue:150]: error on renderStream()

image

File: src/views/StreamReports.vue [hunk 2]
```diff
@@ -155,9 +155,14 @@ export default class StreamReports extends Vue {
   }
 
   edit(branchName: string) {
```
Comment (ro-sa) [inline src/views/StreamReports.vue:157]: the button on the card still allows for edits and it redirects to the Assessment full page, but it seems to not load. It also looks like there are two spinners shown (the circular one and the one in the background coming from the model)

image

**Review by ro-sa — COMMENTED** [review]
LE: superseded comment. `npm i` wasn't run.

the viewer doesn't load on my local, so can't really test much. I've added a couple of comments. Please address those and I'll do a full review once those are fixed. Thanks.

**Comment by ro-sa** [comment]
Screenshot 2022-12-20 at 11 43 45

we are showing two kinds of spinners for the new assessment. we should only keep the one for the model (the bar one).

**Comment by ro-sa** [comment]
Screenshot 2022-12-20 at 11 43 12
.githubusercontent.com/79453579/208661105-c6a6652f-88ab-49cf-a853-8d3ad9208a33.png)


it looks like the volume shows as NaN when editing models. checking the next step in the assessment stepper shows that A1-A3 is 0, so the value is no longer computed correctly.

File: src/components/core/VersionUpdateDialog.vue [hunk 3]
```diff
@@ -56,15 +56,21 @@ export default class VersionUpdateDialog extends Vue {
...
+  ];
 
   fixes: { title: string; subtitle: string }[] = [
     {
       title: "Bug on stream with large number of branches",
       subtitle: `A bug causing issues with streams that have a large number of branches has been fixed.
-                NOTE: if you're stream has more than 100 branches then issues may occur`
-    }
-  ]
+                NOTE: if you're stream has more than 100 branches then issues may occur`,
```
Comment (ro-sa) [inline src/components/core/VersionUpdateDialog.vue:71]: your*

File: src/components/core/VersionUpdateDialog.vue [hunk 4]
```diff
@@ -56,15 +56,21 @@ export default class VersionUpdateDialog extends Vue {
   @Prop() dialog!: boolean;
   @Prop() version!: string;
 
-  features: { title: string; subtitle: string }[] = [];
+  features: { title: string; subtitle: string }[] = [
+    {
+      title: "Updated Speckle Viewer",
+      subtitle: `Updated the Speckle Viewer to include some of the latest features.
```
Comment (ro-sa) [inline src/components/core/VersionUpdateDialog.vue:62]: Updated the Speckle Viewer to include some of the latest features. It also speeds up the assessment and allows for running assessments on larger models.

**Review by ro-sa — COMMENTED** [review]
good implementation of the viewer. it is noticeably faster and smoother and it just looks better. please address the comments before I can approve.

**Comment by Exilliar** [comment]
> Screenshot 2022-12-20 at 11 43 45
> 
> we are showing two kinds of spinners for the new assessment. we should only keep the one for the model (the bar one).

I'm thinking that it may be better to leave the spinner one. There's some loading that needs to happen after the model has been loaded (loading in the objects from the model and grouping them into Speckle groups) which can take a little bit of time on larger models. I feel like the spinner helps to get this point aross a little better?

**Comment by Exilliar** [comment]
> Screenshot 2022-12-20 at 11 43 12 Screenshot 2022-12-20 at 11 54 50
> 
> it looks like the volume shows as NaN when editing models. checking the next step in the assessment stepper shows that A1-A3 is 0, so the value is no longer computed correctly.

Good spot! Hadn't tested any Rhino models, will have to see what's going on there

File: src/components/core/VersionUpdateDialog.vue (hunk 3 above)
Comment (Exilliar) [reply src/components/core/VersionUpdateDialog.vue:71]: I think that we can probalby remove this note now as #179 should have fixed this issue

**Comment by ro-sa** [comment]
> 

ok, fair point.

could we add an overlay or fade the background when the spinner is still on?

**Review by ro-sa — APPROVED** [review]

**Pull request merged**